Sources/StoryBB/Template.php

```diff
@@ -260,6 +260,24 @@ public static function get_debug_info()
 	{
 		return self::$debug;
 	}
+
+	public static function get_theme_id(string $template_type, string $template_name): int {
+		global $settings;
+
+		if (!in_array($template_type, ['partials', 'templates', 'layouts']))
+			fatal_error('Invalid template type: ' . $template_type);
+
+		if ($settings['default_theme_dir'] == $settings['theme_dir'])
+			return 1; // Default theme.
+
+		if (!file_exists($settings['theme_dir'] . '/' . $template_type))
```

Comment:
do you need this when you are returning 1 by default and you are checking for file_exists on the specific file a line lower?

Reply:
If you try to do file_exists on a file in a folder that doesn't exist, it can throw an error that you otherwise might not be able to catch correctly, so here we have to start by checking the folder exists. If the folder doesn't exist, we don't have to worry about a specific file in it and we save ourselves throwing a warning (note: warnings are not exceptions, the only way to suppress a warning is to turn off error handling and turn it back on again afterwards, precisely because they're not exceptions)

If the folder does exist, we have to check the specific file.